SFML/SFML
# [WIP] Implement wrapper for loading libraries on Win32

## Description

This change adds private `LibraryLoader` utility class that automates loading and unloading Win32 dynamic libraries within it's scope.

The library is loaded when object is created and released upon it's destruction.
The implementation of `loadEntryPoint` routine in current Vulkan library loading wrapper implementation was very nice. After trying a few different approaches I've decided that current function definition is most elegant for almost all usage scenarios that we have at the moment.

I've updated Window implementation, Joystick implementation and Vulkan implementation to use new LibraryLoader class.

The documentation for `LibraryLoader ` is not completed yet. I would like to get feedback about this change before writing the documentation.

I've implemented no automated tests for implemented class yet. Please let me know if I should add tests. 

I was using Windows 10 21H1 build 19043.1348 and Visual Studio 2019 (16.11.7) with Windows SDK 10.0.22000.0 to develop and test this change.

## Tasks

* [NO] Tested on Linux
* [YES] Tested on Windows
* [NO] Tested on macOS
* [NO] Tested on iOS
* [NO] Tested on Android

## How to test this PR?

I've manually tested example applications `joystick`, `opengl`, `vulkan`, `win32` and `window` to make sure that the change isn't breaking anything.
There are no visible changes or added features.

```
